stan/math/prim/mat.hpp

```diff
@@ -46,6 +46,8 @@
 #include <stan/math/prim/mat/err/constraint_tolerance.hpp>
 #include <stan/math/prim/mat/err/validate_non_negative_index.hpp>
 
+#include <stan/math/prim/mat/vectorize/apply_scalar_unary.hpp>
```

Comment:
Okay. I tried changing the order for the prim version of `apply_scalar_unary.hpp` and the tests ran fine for the vectorized tests of `exp` and `foo`. However, when I tried changing the order for the rev version of `apply_scalar_unary`, the vectorized tests of `foo` crashed.